client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -221,5 +232,52 @@ private void validateStreamCut(Map<Stream, StreamCut> streamCuts) {
                }
            });
        }
-   }
+    }
+
+    private static class ReaderGroupConfigSerializer
+            extends VersionedSerializer.WithBuilder<ReaderGroupConfig, ReaderGroupConfigBuilder> {
+        @Override
+        protected ReaderGroupConfigBuilder newBuilder() {
+            return builder();
+        }
+
+        @Override
+        protected byte getWriteVersion() {
+            return 0;
+        }
+
+        @Override
+        protected void declareVersions() {
+            version(0).revision(0, this::write00, this::read00);
+        }
+
+        private void read00(RevisionDataInput revisionDataInput, ReaderGroupConfigBuilder builder) throws IOException {
+            builder.automaticCheckpointIntervalMillis(revisionDataInput.readLong());
+            builder.groupRefreshTimeMillis(revisionDataInput.readLong());
+            ElementDeserializer<Stream> keyDeserializer = in -> Stream.of(in.readUTF());
+            ElementDeserializer<StreamCut> valueDeserializer = in -> StreamCut.fromBytes(ByteBuffer.wrap(in.readArray()));
+            builder.startFromStreamCuts(revisionDataInput.readMap(keyDeserializer, valueDeserializer));
+            builder.endingStreamCuts(revisionDataInput.readMap(in -> Stream.of(in.readUTF()), valueDeserializer));
```

Comment:
can you use `keyDeserializer` here? they look similar and you do use it in the `write00` method